ludwig/utils/schema_utils.py

```diff
@@ -1,29 +1,153 @@
+import re
 from dataclasses import field
+from typing import Dict, List, Tuple, Type, Union
 
 import marshmallow_dataclass
-from marshmallow import fields, validate, ValidationError
+from marshmallow import fields, Schema, validate, ValidationError
+from marshmallow_jsonschema import JSONSchema as js
+from pytkdocs import loader as pytkloader
 
 from ludwig.modules.reduction_modules import reduce_mode_registry
 from ludwig.utils.torch_utils import initializer_registry
 
+restloader = pytkloader.Loader(docstring_style="restructured-text")
+googleloader = pytkloader.Loader(docstring_style="google")
 
-def InitializerOptions(default=None):
+
+def create_cond(if_pred: Dict, then_pred: Dict):
+    """Returns a JSONSchema conditional for the given if-then predicates."""
+    return {
+        "if": {"properties": {k: {"const": v} for k, v in if_pred.items()}},
+        "then": {"properties": {k: v for k, v in then_pred.items()}},
+    }
+
+
+def get_fully_qualified_class_name(cls):
```

Comment:
Worth moving this docstring stuff to a separate file?

Reply:
Keeping it in this file seems fine for now (imo).